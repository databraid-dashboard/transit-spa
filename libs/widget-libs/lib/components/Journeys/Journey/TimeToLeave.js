var _jsxFileName = '../../src/components/Journeys/Journey/TimeToLeave.jsx',
    _this = this;

import React from 'react';
import PropTypes from 'prop-types';

import { Icon, List } from 'semantic-ui-react';
import Timer from './Timer';

export function timerExpired() {
  return React.createElement(
    'div',
    {
      __source: {
        fileName: _jsxFileName,
        lineNumber: 8
      },
      __self: this
    },
    'RUN!!!'
  );
}

var TimeToLeave = function TimeToLeave(_ref) {
  var timeToLeaveInSeconds = _ref.timeToLeaveInSeconds;
  return React.createElement(
    List.Item,
    {
      __source: {
        fileName: _jsxFileName,
        lineNumber: 12
      },
      __self: _this
    },
    React.createElement(Icon, { name: 'hourglass half', size: 'large', __source: {
        fileName: _jsxFileName,
        lineNumber: 13
      },
      __self: _this
    }),
    React.createElement(
      List.Content,
      {
        __source: {
          fileName: _jsxFileName,
          lineNumber: 14
        },
        __self: _this
      },
      React.createElement(
        List.Header,
        {
          __source: {
            fileName: _jsxFileName,
            lineNumber: 15
          },
          __self: _this
        },
        'leave in'
      ),
      React.createElement(Timer, { seconds: timeToLeaveInSeconds, onComplete: timerExpired, __source: {
          fileName: _jsxFileName,
          lineNumber: 16
        },
        __self: _this
      })
    )
  );
};

TimeToLeave.propTypes = {
  timeToLeaveInSeconds: PropTypes.number.isRequired
};

TimeToLeave.defaultProps = {
  timeToLeaveInSeconds: 5
};

export default TimeToLeave;