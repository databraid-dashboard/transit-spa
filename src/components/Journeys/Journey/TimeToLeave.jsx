import React from 'react';
import PropTypes from 'prop-types';

import { Icon, List } from 'semantic-ui-react';
import Timer from './Timer';

export function timerExpired() {
  return <div>RUN!!!</div>;
}

const TimeToLeave = ({ timeToLeaveInSeconds }) =>
  (<List.Item>
    <Icon name="hourglass half" size="large" />
    <List.Content>
      <List.Header>leave in</List.Header>
      <Timer seconds={timeToLeaveInSeconds} onComplete={timerExpired} />
    </List.Content>
  </List.Item>);

TimeToLeave.propTypes = {
  timeToLeaveInSeconds: PropTypes.number.isRequired,
};

TimeToLeave.defaultProps = {
  timeToLeaveInSeconds: 5,
};

export default TimeToLeave;
