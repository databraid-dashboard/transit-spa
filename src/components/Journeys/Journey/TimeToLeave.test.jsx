import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import TimeToLeave, { timerExpired } from './TimeToLeave';

describe('TimeToLeave', () => {
  it('should render list item', () => {
    const seconds = 120;

    const component = mount(<TimeToLeave timeToLeaveInSeconds={seconds} />);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('has timeToLeaveInSeconds prop as a number of seconds', () => {
    const seconds = 120;

    const component = mount(<TimeToLeave timeToLeaveInSeconds={seconds} />);
    expect(component.props().timeToLeaveInSeconds).toEqual(seconds);
  });

  it('has timerExpired callback', () => {
    expect(timerExpired()).toEqual(<div>RUN!!!</div>);
  });
});
