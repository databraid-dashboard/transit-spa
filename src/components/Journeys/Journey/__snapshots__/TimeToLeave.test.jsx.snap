// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeToLeave should render list item 1`] = `
<TimeToLeave
  timeToLeaveInSeconds={120}
>
  <ListItem>
    <div
      className="item"
      onClick={[Function]}
      role="listitem"
    >
      <Icon
        as="i"
        name="hourglass half"
        size="large"
      >
        <i
          aria-hidden="true"
          className="hourglass half large icon"
        />
      </Icon>
      <ListContent>
        <div
          className="content"
        >
          <ListHeader>
            <div
              className="header"
            >
              leave in
            </div>
          </ListHeader>
          <Timer
            onComplete={[Function]}
            seconds={120}
          >
            <div>
              02
              :
              00
            </div>
          </Timer>
        </div>
      </ListContent>
    </div>
  </ListItem>
</TimeToLeave>
`;
